lib/sshutils/server.go

```diff
@@ -810,16 +837,43 @@ func (c *ConnectionWrapper) Read(b []byte) (int, error) {
 			skip = payloadBoundary + 1
 		}
 	}
-	c.upstreamReader = io.MultiReader(bytes.NewBuffer(buff[skip:]), c.Conn)
+	if bytes.HasPrefix(buff, multiplexer.ProxyV2Prefix) {
+		reader := bufio.NewReader(io.MultiReader(bytes.NewBuffer(buff), c.Conn))
+
+		newProxyLine, err := multiplexer.ReadProxyLineV2(reader)
+		if err != nil {
+			return 0, trace.Wrap(err)
+		}
+		if newProxyLine.IsSigned() && c.caGetter != nil {
+			err = newProxyLine.VerifySignature(context.Background(), c.caGetter, c.clusterName, c.clock)
+			if err != nil {
+				return 0, trace.Wrap(err)
+			}
+			if newProxyLine != nil {
```

Comment:
Is this necessary? in order to call `newProxyLine.IsSigned()`, `newProxyLine` must be non-nil

Reply:
Nice catch 👍 
I actually wanted to check if proxyline is verified here, because I also want to ignore couple of errors